pkg/localmetrics/localmetrics.go

```diff
@@ -141,7 +142,7 @@ func NewMetricsCollector(store cache.Cache) *MetricsCollector {
 			Help:        "Distribution of the number of seconds a Reconcile takes, broken down by controller",
 			ConstLabels: prometheus.Labels{"name": operatorName},
 			Buckets:     []float64{0.001, 0.01, 0.1, 1, 5, 10, 20},
-		}, []string{"controller", "error"}),
+		}, []string{"controller", "error", "aws_error"}),
```

Comment:
I don't love the idea of making an aws-specific label here. Why not just make it something like `error_code` and use a special value like `{OTHER}` when it's one we don't have special parsing for (which would include AWS errors that didn't cast). And you could also include the a-a-o errors with specific class names using the `%T` we talked about last week.

For that matter, you could do all of the above in the existing `error` dimension. It's not _exactly_ backward compatible with the existing bool, but I don't think that much matters (and it's copacetic wrt data type, since it's a string in prom anyway).

Reply:
Per PR review: make two dimensions like `error_source` ("aws", "k8s", etc) and `error_code` (or leave it as `error`) for the specifics.